shell/src/main/java/alluxio/cli/LogLevel.java

```
@@ -0,0 +1,257 @@
+              + " you want to get or set level.")
+          .build();
+  private static final String LEVEL_OPTION_NAME = "level";
+  private static final Option LEVEL_OPTION =
+      Option.builder()
+          .required(false)
+          .longOpt(LEVEL_OPTION_NAME)
+          .hasArg(true)
+          .desc("The log level to be set.").build();
+  private static final Options OPTIONS = new Options()
+      .addOption(TATGET_OPTION)
+      .addOption(LOG_NAME_OPTION)
+      .addOption(LEVEL_OPTION);
+
+  /**
+   * Prints the help message.
+   *
+   * @param message message before standard usage information
+   */
+  public static void printHelp(String message) {
+    System.err.println(message);
+    HelpFormatter help = new HelpFormatter();
+    help.printHelp(LOG_LEVEL, OPTIONS, true);
+  }
+
+  /**
+   * Implements log level setting and getting.
+   *
+   * @param args list of arguments contains target, logName and level
+   * @return 0 on success, 1 on failures
+   * @exception ParseException if there is an error in parsing
+   * @exception IOException if the rpc server cannot reached
+   */
+  public static int logLevel(String... args) throws ParseException, IOException {
+    CommandLineParser parser = new DefaultParser();
+    CommandLine cmd = parser.parse(OPTIONS, args, true /* stopAtNonOption */);
+
+    List<TargetInfo> targets = parseOptTarget(cmd);
+    String logName = parseOptLogName(cmd);
+    String level = parseOptLevel(cmd);
+
+    for (TargetInfo targetInfo : targets) {
+      if (handleTarget(targetInfo, logName, level)) {
+        return 1;
+      }
+    }
+    return 0;
+  }
+
+  private static List<TargetInfo> parseOptTarget(CommandLine cmd) throws IOException {
+    String[] targets;
+    if (cmd.hasOption(TARGET_OPTION_NAME)) {
+      String argTarget = cmd.getOptionValue(TARGET_OPTION_NAME);
+      if (StringUtils.isBlank(argTarget)) {
+        targets = new String[]{"master", "workers"};
```

This seems like a mistake by the user - we should throw an exception